### PR TITLE
Cleanup and CLI

### PR DESCRIPTION
**Pull request by sourencho**

### CODE REVIEW

File: test/crawler-test.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        assert.strictEqual(error.message, "Invalid string");
+      })
+      .then(done, done);
+    });
+    // ip address
+    it('Should throw an error when given undefined ip address', function(done) {
+      var crawler = new Crawler(100);
+      var entryIP = undefined;
+
+      crawler.getCrawl(entryIP).catch(function(error) {})
```
Comment (wilsonianb) [inline test/crawler-test.js]: Add `assert.strictEqual(error.message, "Invalid string");`

**Comment by wilsonianb** [comment]
It looks like `src/crawler.js` was moved to `src/lib/crawlers.js`, but the original wasn't deleted.
Also the new one appears to be missing the changes from this awesome commit: https://github.com/ripple/rippled-network-crawler/commit/319304424880180a5dbb2445b0fb81689960815

File: src/graphify.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+'use strict';
+var Sequelize = require('sequelize');
+var rc_util = require('./lib/utility.js');
+var modelsFactory = require('./lib/models.js');
+var _ = require('lodash');
+
+function getCrawlById(dbUrl, id, logsql) {
```
Comment (wilsonianb) [inline src/graphify.js]: `getCrawlById` is currently duplicated in `src/info.js`
What do you think about defining it in a single place?

File: src/lib/utility.js [hunk 3]
```diff
@@ -218,15 +218,55 @@ module.exports = {
   /*
   * @param {Object} raw crawl
   * @return {Array} ipp
-  * Takes a raw crawl and returns an 
+  * Takes a raw crawl and returns an
   * array of the ipps which were crawled
   */
-  getIpps: function(nodes) {
+  getCrawledIpps: function(nodes) {
```
Comment (wilsonianb) [inline src/lib/utility.js:226]: `getCrawledIpps` doesn't appear to be used anywhere

File: src/selective.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+var src = require('./program');
+
+module.exports = function(ipps, commander) {
+  var logger = commander.quiet ? {log: _.noop, error: _.noop} : console;
+  var maxRequests = commander.max ? parseInt(commander.max, 10) : 100;
+  var crawler = new Crawler(maxRequests, logger);
+  crawler.getSelCrawl(ipps)
+    .then(function(response) {
+      if (commander.store) {
+        src.store(response, commander.store, commander.logsql);
```
Comment (wilsonianb) [inline src/selective.js:13]: Another case where `commander.store` should maybe be `commander.dbUrl`

File: src/store.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+var Sequelize = require('sequelize');
+var modelsFactory = require('./lib/models.js');
+
+function saveDB(crawlJson, dbUrl, logsql, onDone) {
```
Comment (wilsonianb) [inline src/store.js:5]: Consider returning a `Promise` instead of taking a callback.

File: test/crawler-test.js [hunk 6]
```diff
@@ -26,57 +25,56 @@ describe('Crawler', function() {
...
+    });
     // ip address
     it('Should throw an error when given undefined ip address', function(done) {
-      var crawler = new Crawler(100)
-      var entryIP = undefined
+      var crawler = new Crawler(100);
+      var entryIP = undefined;
 
-      crawler.getCrawl(entryIP).catch(function(error) {
-        assert.strictEqual(error.message, "Invalid string");
```
Comment (wilsonianb) [inline test/crawler-test.js:56]: Re-add the assert

File: test/crawler-test.js [hunk 7]
```diff
@@ -123,55 +121,53 @@ describe('Crawler', function() {
...
-    });    
+    });
     it('Should throw an error when given any undefined ip address', function(done) {
-      var crawler = new Crawler(100)
-      var ipps = ["192.170.145.70:51235", undefined]
+      var crawler = new Crawler(100);
+      var ipps = ["192.170.145.70:51235", undefined];
 
-      crawler.getSelCrawl(ipps).catch(function(error) {
-        assert.strictEqual(error.message, "Invalid string");
```
Comment (wilsonianb) [inline test/crawler-test.js:149]: Re-add the assert

File: src/lib/utility.js (hunk 3 above)
Comment (sourencho) [inline src/lib/utility.js:226]: it's just an extra function for future use

File: src/graphify.js (hunk 2 above)
Comment (sourencho) [inline src/graphify.js]: I'll move it to src/lib/utility.js

File: src/selective.js (hunk 4 above)
Comment (sourencho) [inline src/selective.js:13]: `dbUrl` is used when we get from the database
`commander.store` is used when we write to the database 

They're different variables. `commander.dbUrl` doesn't exist.
I guess my variable names are somewhat confusing. Sorry about that.

it's all in index.js

**Comment by wilsonianb** [comment]
LGTM :+1:

**Pull request merged**